crates/uv/src/commands/tool/run.rs

```diff
@@ -517,52 +472,149 @@ async fn show_help(
     Ok(())
 }
 
-/// Display a warning if an executable is not provided by package.
-///
-/// If found in a dependency of the requested package instead of the requested package itself, we will hint to use that instead.
-fn warn_executable_not_provided_by_package(
-    executable: &str,
-    from_package: &PackageName,
-    site_packages: &SitePackages,
+/// A set of hints about the packages that provide an executable.
+#[derive(Debug)]
+struct ExecutableProviderHints<'a> {
```

Comment:
Are these changes intended as part of this PR?

Reply:
No, just needed to rebase against the newly rebased `release/070`